src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -48,7 +49,14 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
 
-                if (exeAssembly != null)
+                // in case of SingleFile deployment, Assembly.Location is empty.
+                if (exeAssembly != null && string.IsNullOrEmpty(exeAssembly.Location))
+                {
+                    isSingleFile = true;
+                    HasEntryAssembly = true;
```

Comment:
HasEntryAssembly should be set to true when GetEntryAssembly returns non-null assembly, independent on the single file. 

HasEntryAssembly is used to avoid refreshing the configuration unnecessarily when the application has entry assembly. This should be done for single file too.

Reply:
It will be set to true in that case? Looks like GitHub is not showing the clear delta but if you look at the full diff, it is there.

Reply:
Ah, ok. I have missed it.